### PR TITLE
OSX Default Flags

### PR DESCRIPTION
**Pull request by sroberts**
I haven't tested this yet, but it should work correctly. LaunchDaemons are weird.

### CODE REVIEW

**Comment by osqueryer** [comment]
Can one of the osquery admins reply with "ok to test" to kick off a build...

**Comment by sroberts** [comment]
Oh, and I went and signed the CLA as well.

File: tools/deployment/com.facebook.osqueryd.plist [hunk 1]
```diff
@@ -9,7 +9,7 @@
   <key>Label</key>
   <string>com.facebook.osqueryd</string>
   <key>Program</key>
-        <string>/usr/local/bin/osqueryd</string>
+    <string>/usr/local/bin/osqueryd --flagfile="/var/osquery/osquery.flags"</string>
```
Comment (theopolis) [inline tools/deployment/com.facebook.osqueryd.plist]: We should probably change `<key>Program</key>` to `<key>ProgramArguments</key>` and use:

```xml
  <key>ProgramArguments</key>
  <array>
    <string>/usr/local/bin/osqueryd</string>
    <string>--flagfile="/var/osquery/osquery.flags"</string>
  </array>
```

I don't think there's <i>really</i> a difference, but when you go past an argv[0] ProgramArguments is [recommended](https://developer.apple.com/library/mac/documentation/MacOSX/Conceptual/BPSystemStartup/Chapters/CreatingLaunchdJobs.html).

Comment (sroberts) [inline tools/deployment/com.facebook.osqueryd.plist]: @theopolis Makes sense to me. I kept it simple, but didn't check out the spec. Certainly should have.

**Comment by sroberts** [comment]
@theopolis Updated as discussed.

**Comment by theopolis** [comment]
Looks good! If you don't mind squashing the commits into 1, I'll merge ASAP!

**Comment by sroberts** [comment]
Did that work? Honestly this is only the second time I've done this and I just followed a guide that made sense.

**Comment by theopolis** [comment]
Try:
```
git reset --hard 12588002799155ce0c45d910ebe80831b6d97959
git push -f
```

**Comment by sroberts** [comment]
Done and done.

**Comment by sroberts** [comment]
:metal: @theopolis Thanks for the help! Glad to see this in.

**Pull request merged**